pkg/cache/cache_test.go

```diff
@@ -69,185 +125,180 @@ var _ = Describe("Informer Cache", func() {
 		})
 
 		It("should be able to get objects that haven't been watched previously", func() {
-			By("Creating the cache")
-			reader, err := cache.New(cfg, cache.Options{})
+			By("getting the Kubernetes service")
+			svc := &kcorev1.Service{}
+			svcKey := client.ObjectKey{Namespace: "default", Name: "kubernetes"}
+			Expect(informerCache.Get(context.Background(), svcKey, svc)).NotTo(HaveOccurred())
+
+			By("verifying that the returned service looks reasonable")
+			Expect(svc.Name).To(Equal("kubernetes"))
+			Expect(svc.Namespace).To(Equal("default"))
+		})
+
+		It("should support filtering by labels", func() {
+			By("listing pods with a particular label")
+			// NB: each pod has a "test-label" equal to the pod name
+			out := kcorev1.PodList{}
+			Expect(informerCache.List(context.Background(), client.InNamespace(testNamespaceTwo).
+				MatchingLabels(map[string]string{"test-label": "test-pod-2"}), &out)).NotTo(HaveOccurred())
+
+			By("verifying the returned pods have the correct label")
+			Expect(out.Items).NotTo(BeEmpty())
```

Comment:
This can also be written:

```go
Expect(out.Items).To(And(
  HaveLen(1),
  HaveKeyWithValue("test-label", "test-pod-2"),
))
```

Reply:
No, because we're testing actual.Labels, so that code won't work

Reply:
ah right,

so just `Expect(actual.Labels).To(HaveKeyWithValue("test-label", "test-pod-2"))`